src/CommunityToolkit/Xamarin.CommunityToolkit/Effects/BarStyle/PlatformBarStyle.android.cs

```diff
@@ -0,0 +1,164 @@
+		{
+			if (Build.VERSION.SdkInt < BuildVersionCodes.M)
+				return;
+
+			SetBarAppearance((barAppearanceLegacy, barAppearance) =>
+			{
+				switch (style)
+				{
+					case StatusBarStyle.Default:
+					case StatusBarStyle.LightContent:
+						barAppearanceLegacy |= (StatusBarVisibility)SystemUiFlags.LightStatusBar;
+						barAppearance |= WindowInsetsControllerAppearance.LightStatusBars;
+						break;
+					case StatusBarStyle.DarkContent:
+						barAppearanceLegacy &= ~(StatusBarVisibility)SystemUiFlags.LightStatusBar;
+						barAppearance &= ~WindowInsetsControllerAppearance.LightStatusBars;
+						break;
+				}
+				return (barAppearanceLegacy, barAppearance);
+			});
+		}
+
+		public void SetNavigationBarColor(Color color)
+		{
+			if (Build.VERSION.SdkInt < BuildVersionCodes.M)
+				return;
+
+			Activity?.Window.SetNavigationBarColor(color);
+		}
+
+		public void SetNavigationBarStyle(NavigationBarStyle style)
+		{
+			if (Build.VERSION.SdkInt < BuildVersionCodes.M)
+				return;
+
+			SetBarAppearance((appearanceLegacy, appearance) =>
+			{
+				switch (style)
+				{
+					case NavigationBarStyle.Default:
+					case NavigationBarStyle.LightContent:
+						appearanceLegacy |= (StatusBarVisibility)SystemUiFlags.LightNavigationBar;
+						appearance |= WindowInsetsControllerAppearance.LightNavigationBars;
+						break;
+					case NavigationBarStyle.DarkContent:
+						appearanceLegacy &= ~(StatusBarVisibility)SystemUiFlags.LightNavigationBar;
+						appearance &= ~WindowInsetsControllerAppearance.LightNavigationBars;
+						break;
+				}
+				return (appearanceLegacy, appearance);
+			});
+		}
+
+		static void SetBarAppearance(Func<StatusBarVisibility, WindowInsetsControllerAppearance, (StatusBarVisibility, WindowInsetsControllerAppearance)> updateAppearance)
+		{
+			var currentWindow = Activity?.Window;
+			if (currentWindow == null)
+				return;
+
+			StatusBarVisibility appearanceLegacy = 0;
+			WindowInsetsControllerAppearance appearance = 0;
+			if ((int)Build.VERSION.SdkInt < 30)
+			{
+#pragma warning disable CS0618 // Type or member is obsolete. Using new API for Sdk 30+
+				appearanceLegacy = currentWindow.DecorView.SystemUiVisibility;
+#pragma warning restore CS0618 // Type or member is obsolete
+			}
+			else
+			{
+				appearance = (WindowInsetsControllerAppearance)currentWindow.InsetsController.SystemBarsAppearance;
+			}
+
+			(appearanceLegacy, appearance) = updateAppearance(appearanceLegacy, appearance);
+
+			if ((int)Build.VERSION.SdkInt < 30)
+			{
+#pragma warning disable CS0618 // Type or member is obsolete. Using new API for Sdk 30+
+				currentWindow.DecorView.SystemUiVisibility = appearanceLegacy;
+#pragma warning restore CS0618 // Type or member is obsolete
+			}
+			else
+			{
+				currentWindow.InsetsController.SetSystemBarsAppearance((int)appearance, (int)appearance);
+			}
+		}
+
+		static FormsAppCompatActivity Activity => App.Context.GetActivity() as FormsAppCompatActivity;
```

Comment:
Could someone please suggest how to het the activity here?
This doesn't work

Reply:
Since you are inside the `Effect` you can grab the Context from the `Control` or `Container` property. Can you try it? If that doesn't solve your trouble our implementation on PopUp PR will solve.

Reply:
Unfortunately both `Container.Context.GetActivity()` and `Control.Context.GetActivity()` are `null`. Waiting on PopUp PR then

Reply:
I think it's fine to cast the `Control.Context` to Activity.

Reply:
It happened to me some strange behavior, not taking modifications into account when debugging with breakpoint on it, could you try to clean and rebuild `Container.Context.GetActivity()` should work.

Reply:
@maxkoshevoi I fixed this issue. Now the effect is working